### PR TITLE
Fix: filetree-sync-data-with-codeEditor

### PR DESCRIPTION
**Pull request by Captain-Tab**
- [x] fix: add background color to selected folder
- [x] fix: keep copy file target accuracy 
- [x] fix:  click tab of code-editor will open folder of fileTree

### CODE REVIEW

**Comment by iNorthIsle** [comment]
The file changes look like you need to rebase

**Review by iNorthIsle — CHANGES_REQUESTED** [review]
The file changes look like you need to rebase and resolve the conflict

**Comment by Captain-Tab** [comment]
> The file changes look like you need to rebase and resolve the conflict

> rebase completed

File: packages/filetree/src/index.js [hunk 1]
```diff
@@ -140,13 +144,36 @@ const FileTree = forwardRef(({ projectManager, onSelect, move, copy, initialPath
...
     })
 
+    React.useImperativeHandle(ref, () => ({
+      setActive(key) {
+        handleSetActive(key)
+      },
+      resetActive() {
+        setSelectedKeys([])
+        setSelectNode(null)
+      },
```
Comment (iNorthIsle) [inline packages/filetree/src/index.js:154]: how about:
```suggestion
      setActive(key) {
           if(key) {
               handleSetActive(key)
               return
           }
          setSelectedKeys([])
          setSelectNode(null)
      },
```

File: packages/filetree/src/index.js [hunk 2]
```diff
@@ -303,13 +299,41 @@ const FileTree = forwardRef(({ projectManager, onSelect, move, copy, initialPath
...
+            }
+            stop = true
+          }
+          return stop
+        }
+        !selectedKeys.includes(activeKey) && setSelectedKeys([activeKey])
+        !selectNode.path !== activeKey && travelTree(...treeData, findNodeByKey, activeKey)
+      }
+
+      const delayFunc = (func, delayTime) => {
```
Comment (iNorthIsle) [inline packages/filetree/src/index.js:323]: it's not necessary to abstract the function

File: packages/workspace/src/components/Workspace.js [hunk 3]
```diff
@@ -99,23 +100,20 @@ export default class Workspace extends Component {
 
   tabFromPath = (filePath, remote, pathInProject) => ({ path: filePath, key: filePath, remote, pathInProject })
 
+  setFileTreeActive (path) {
+    path ? this.filetree.current.setActive(path) : this.filetree.current.resetActive()
```
Comment (iNorthIsle) [inline packages/workspace/src/components/Workspace.js:104]: then you can:
```suggestion
    this.filetree.current.setActive(path ? path : null)
```

**Review by iNorthIsle — APPROVED** [review]

**Pull request merged**